components/json/BUILD.gn

```diff
@@ -23,10 +23,25 @@ component("json") {
 source_set("brave_json_unit_tests") {
   testonly = true
   sources = [ "//brave/components/json/json_parser_unittest.cc" ]
+
   deps = [
     "//base",
     "//base/test:test_support",
     "//brave/components/json",
     "//testing/gtest",
   ]
+
+  if (!is_android) {
+    # Android seems to have some issue running these base tests, specially when
+    # using base::DIR_TEST_DATA
+    sources += [
+      "//base/json/json_reader_unittest.cc",
```

Comment:
the problem you're having here is because these tests (like most tests currently in brave/components) are running as part of chrome unit tests. To work correctly this test should be moved into `brave_components_unittest`, but why do we need to run the upstream tests here when we already run them as part of chromium unit tests?

Reply:
I think tests we add for this will go in a brave-specific json unittest, but either way we haven't actually integrated anything into json reader/json writer in this PR so there's no reason to do anything with these at least in this PR?

Reply:
> We need to know if our changes to the visitor are breaking the default behaviour

Aren't we already running `base_unittests` as part of chromium unit tests? In any case the base::DIR_TEST_DATA issue can be fixed by moving this test target to brave_components_unittest target it they belongs anyway.

Reply:
Fixed.